app/services/katello/pulp/repository.rb

```diff
@@ -247,6 +250,29 @@ def index_content_units
           content_type.pulp2_service_class
         end
       end
+
+      def proxy_options
+        if repo.root.http_proxy_policy == RootRepository::NO_DEFAULT_HTTP_PROXY
+          return { proxy_host: '' }
+        end
+
+        proxy = repo.root.http_proxy
+        if repo.root.http_proxy_policy == RootRepository::GLOBAL_DEFAULT_HTTP_PROXY
+          proxy = HttpProxy.default_global_content_proxy
+        end
+
+        if proxy
+          uri = URI(proxy.url)
+          proxy_options = {
+            proxy_host: uri.host,
+            proxy_port: uri.port,
+            proxy_username: proxy.username,
+            proxy_password: proxy.password
+          }
+          return proxy_options
+        end
+        {}
```

Comment:
I think proxy_host should be empty string here, as if there is no global proxy defined, it will pass no option, and thus use the old pulp2 global proxy (which we don't want).